src/frontend/store/graphql.js

```diff
@@ -0,0 +1,32 @@
+export default {
+  namespaced: true,
+  state: () => ({
+    simple: [],
+    best_oa_location: [],
+    first_oa_location: [],
+    oa_locations: [],
+    z_authors: []
+  }),
+  actions: {
+    add ({ commit }, value) {
+      commit('add', { source: value.source, attrs: value.attrs })
+    },
+    update ({ commit }, value) {
+      commit('update', { source: value.source, attrs: value.attrs })
+    },
+    del ({ commit }, value) {
+      commit('del', { source: value.source, attrs: value.attrs })
+    }
+  },
+  mutations: {
+    add (state, value) {
+      state[value.source].push(value.attrs)
+    },
+    update (state, value) {
+      state[value.source] = value.attrs
+    },
+    del (state, value) {
+      state[value.source] = state[value.source].filter(e => e === value)
+    }
```

Comment:
No type check either here or in the actions. Be careful because if `state[value.source]` turns out to be something else than an array, or if `value` is `undefined`, this code will crash.